text/0328-polyglot-assert.md

```diff
@@ -0,0 +1,335 @@
+    },
+  },
+});
+```
+
+#### Counting Resources
+
+This module allows asserting the number of resources of a specific type found
+in a template.
+
+```ts
+inspect.assertResourceCountBeta1('Foo::Bar', 2);
+```
+
+#### Resource Matching
+
+Beyond resource counting, the module also allows asserting that a resource with
+specific properties are present.
+
+The following code asserts that the `Properties` section of a resource of type
+`Foo::Bar` contains the specified properties -
+
+```ts
+inspect.assertResourceBeta1('Foo::Bar', {
+  Foo: 'Bar',
+  Baz: 5,
+  Qux: [ 'Waldo', 'Fred' ],
+});
+```
+
+The same method allows asserting the complete definition of the 'Resource'
+which can be used to verify things other sections like `DependsOn`, `Metadata`,
+`DeletionProperty`, etc.
+
+```ts
+inspect.assertResourceBeta1('Foo::Bar', {
+  Properties: { Foo: 'Bar' },
+  DependsOn: [ 'Waldo', 'Fred' ],
+}, {
+  part: ResourcePartBeta1.COMPLETE_BETA1,
+});
+```
+
+## FAQ
+
+### What are we launching today?
+
+Today, we are launching a new jsii module called `@aws-cdk/assertv2`, that allows
+CDK users to run assertions on the synthesized CloudFormation template.
+
+This module is available to users in all CDK supported languages.
+
+### Why should I use this feature?
+
+You should use this feature if you would like to ensure that your CDK constructs
+and CDK applications render expected CloudFormation resources and properties.
+
+### How is this different from the existing `assert` module?
+
+The two modules provide the same set of features. However, the existing module
+was only available to javascript users.
+The new version launched today is available to users of all CDK supported
+languages available today and any that we will support in the future.
+
+### How do I get started?
+
+This module is available in the languages as per the table below -
+
+| Language              | Pkg Manager | Package                                     |
+| --------------------- | ------------|-------------------------------------------- |
+| typescript/javascript | npm         | `@aws-cdk/assertv2`                         |
+| Java                  | Maven       | `software.amazon.awscdk.assertv2`           |
+| Python                | PyPI        | `aws-cdk.assertv2`                          |
+| .NET                  | NuGet       | `Amazon.CDK.AssertV2`                       |
+| Go                    | -           | `github.com/aws/aws-cdk-go/awscdk/assertv2` |
+
+### Is this feature available in the AWS CDK v2?
+
+This module is available in both the AWS CDK v1 and v2, just like all of our other
+modules.
+
+In CDKv2, this module will be available as part of `aws-cdk-lib`.
+The import statement you will need to use is -
+
+```ts
+import { assertv2 as assert } from 'aws-cdk-lib';
+```
+
+Read more about `aws-cdk-lib` at
+<https://github.com/aws/aws-cdk/tree/master/packages/aws-cdk-lib#readme>.
+
+### Why are all the APIs suffixed with the `Beta1` suffix?
```

Comment:
Can we at least simplify to `Beta` instead of `Beta1` ?

Reply:
I don't think that's what we decided.

On the next iteration, this will be `Beta2` which kind of makes the `Beta` or `Beta1` moot, no?

cc @NetaNir @skinny85

Reply:
I think it's not horrible to suffix the first iteration with `Beta`. If we need a second iteration, we can still use `Beta2`, but if there is never a need for a second iteration, then the preview API looks nicer with just `Beta`.

Reply:
I'd like to keep this `Beta1` unless someone wants to fight me on this. 

Leaving it as `Beta` makes it unclear to the maintainer if the next iteration is `Beta0`, `Beta1`, ...

Reply:
> I'd like to keep this `Beta1` unless someone wants to fight me on this.
> 
> Leaving it as `Beta` makes it unclear to the maintainer if the next iteration is `Beta0`, `Beta1`, ...

I don't think that's a very serious problem, but I don't care enough to fight 🙂.

Reply:
I agree with @nija-at, we want to avoid any ambiguity here. `beta1` < `beta2` in every context, but there are frameworks/products/environments  in which `beta2` < `beta`, does it make sense? Not sure, but let's avoid having to explain it to our users.

Reply:
Since `beta` will be deprecated by the time `beta2` happens, I'm not too concerned about that ambiguity.

But like I said, I don't care too much about this particular aspect.

Reply:
I do, actually.

I hate `Beta1` with a passion. I will fight :)